tests/js/client/aql/vector/aql-vector-nprobe.js

```diff
@@ -0,0 +1,143 @@
+    const seed = 12132390894;
+    const defaultNProbe = 1;
+
+    return {
+        setUpAll: function() {
+            db._createDatabase(dbName);
+            db._useDatabase(dbName);
+
+            collection = db._create(collName, {
+                numberOfShards: 3
+            });
+
+            let docs = [];
+            let gen = randomNumberGeneratorFloat(seed);
+            for (let i = 0; i < 1000; ++i) {
+                const vector = Array.from({
+                    length: dimension
+                }, () => gen());
+                if (i === 0) {
+                    randomPoint = vector;
+                }
+                docs.push({
+                    vector
+                });
+            }
+            collection.insert(docs);
+
+            collection.ensureIndex({
+                name: "vector_l2",
+                type: "vector",
+                fields: ["vector"],
+                inBackground: false,
+                params: {
+                    metric: "l2",
+                    dimension: dimension,
+                    nLists: 500,
+                    trainingIterations: 10,
+                },
+            });
+        },
+
+        tearDownAll: function() {
+            db._useDatabase("_system");
+            db._dropDatabase(dbName);
+        },
+
+        testApproxL2WithOneProbeSameAsDefault: function() {
+            const queryWithoutNProbe =
+                "FOR d IN " +
+                collection.name() +
+                " SORT APPROX_NEAR_L2(d.unIndexedVector, @qp) LIMIT 5 RETURN {key: d._key}";
+            const queryWithNProbe =
+                "FOR d IN " +
+                collection.name() +
+                " SORT APPROX_NEAR_L2(d.unIndexedVector, @qp, {nProbe: 1}) " +
+                "LIMIT 5 RETURN {key: d._key}";
+
+            const bindVars = {
+                qp: randomPoint,
+            };
+
+            const resultsWithoutNProbe = db._query(queryWithoutNProbe, bindVars).toArray();
+            const resultsWithNProbe = db._query(queryWithoutNProbe, bindVars).toArray();
+            assertEqual(resultsWithoutNProbe, resultsWithNProbe);
+        },
+
+        testApproxL2WithNProbe: function() {
+            const queryWithoutNProbe =
+                "FOR d IN " +
+                collection.name() +
+                " SORT APPROX_NEAR_L2(d.unIndexedVector, @qp) LIMIT 5 RETURN {key: d._key}";
+            const queryWithNProbe =
+                "FOR d IN " +
+                collection.name() +
+                " SORT APPROX_NEAR_L2(d.unIndexedVector, @qp, {nProbe: 100}) " +
+                "LIMIT 5 RETURN {key: d._key}";
+
+            const bindVars = {
+                qp: randomPoint,
+            };
+
+            const resultsWithoutNProbe = db._query(queryWithoutNProbe, bindVars).toArray();
+            const resultsWithNProbe = db._query(queryWithoutNProbe, bindVars).toArray();
+            assertNotEqual(resultsWithoutNProbe, resultsWithNProbe);
+        },
```

Comment:
I would expect the search parameters to be reflected in the plan. (also maybe consider adding them to the explain output) and in this test, we should assert, that the parameter is in the plan as we expect it.

Reply:
Yes that is true, will do that!